docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -127,17 +151,16 @@ Suggested and hypothetical configuration for the device service reading filters
 
 ``` toml
 [Writable.Filters]
-    ExecutionOrder = "FilterByValueInRange, FilterByResourceNamesMatch"
+    ExecutionOrder = "FilterByResourceNamesMatch, FilterByValue"
     [Writable.Filter.Functions.FilterByResourceNamesMatch]
         [Writable.Filter.Functions.FilterByResourceNamesMatch.Parameters]
-            DeviceNames = "Random-Float-Device,Random-Integer-Device"
-            FilterOut = "false"
-    [Writable.Filter.Functions.FilterByValueInRange]
-        [Writable.Filter.Functions.FilterByValueInRange.Parameters]
-            Min = 100
-            Max = 200
+            FilterValues = "Int32"
+            FilterOps ="eq"
+    [Writable.Filter.Functions.FilterByValue]
+        [Writable.Filter.Functions.FilterByValue.Parameters]
```

Comment:
You still need to know which resource this applies to. You don't want to filter all reading for this range, just ones for a specific Resource. Add `TargetResourceName = "Int64"` to your example here and struct above. This property will be left blank/ignored in the case when filtering for `ResourceName` as the target values will be in `FilterValues`.

Reply:
Lenny - I had provide for the filter by resource name by applying the target resource names in the values array.  But I think you are suggesting I may want to do both: filter for particular resources with particular values.  Which makes sense.  I have added that into the structure and configuration suggestions.